analytic_engine/src/instance/open.rs

```diff
@@ -167,58 +156,21 @@ impl Instance {
             None => return Ok(None),
         };
 
-        let (tx, rx) = oneshot::channel();
-        let cmd = RecoverTableCommand {
-            space,
-            table_data: table_data.clone(),
-            tx,
-            replay_batch_size: self.replay_batch_size,
-        };
-
-        // Send recover request to write worker, actual works done in
-        // Self::recover_table_from_wal()
-        write_worker::process_command_in_write_worker(cmd.into_command(), &table_data, rx)
-            .await
-            .context(OperateByWriteWorker {
-                space_id: table_data.space_id,
-                table: &table_data.name,
-                table_id: table_data.id,
-            })
-    }
-
-    /// Recover the table data.
-    ///
-    /// Return None if the table data does not exist.
-    pub async fn process_recover_table_command(
-        self: &Arc<Self>,
-        worker_local: &mut WorkerLocal,
-        space: SpaceRef,
-        table_data: TableDataRef,
-        replay_batch_size: usize,
-    ) -> Result<Option<TableDataRef>> {
-        if let Some(exist_table_data) = space.find_table_by_id(table_data.id) {
-            warn!("Open a opened table, table:{}", table_data.name);
-            return Ok(Some(exist_table_data));
-        }
-
         let read_ctx = ReadContext {
-            batch_size: replay_batch_size,
+            batch_size: self.replay_batch_size,
             ..Default::default()
         };
 
-        self.recover_table_from_wal(
-            worker_local,
-            table_data.clone(),
-            replay_batch_size,
-            &read_ctx,
-        )
-        .await
-        .map_err(|e| {
-            error!("Recovery table from wal failed, table_data:{table_data:?}, err:{e}");
-            space.insert_open_failed_table(table_data.name.to_string());
-            e
-        })?;
+        self.recover_table_from_wal(table_data.clone(), self.replay_batch_size, &read_ctx)
```

Comment:
This result in  using meta runtime to open table, which may block heartbeat to meta.

Reply:
I've changed the runtime of `meta_event_service` to `default_runtime`.